feat(kcard): component reskin [KHCP-8971] (#1804)
Kong/kongponents: docs/components/card.md

# Card

> KCard is a styled container that organizes together related content and actions.

Reviewer: ```suggestion
KCard is a styled container that organizes related content and actions.
```

<KCard>
  <template #title>
    Card Title
  </template>
  <template #actions>
    <KDropdown
      :items="[
        { label: 'Props', to: { path: '/components/card#props' } },
        { label: 'Slots', to: { path: '/components/card#slots' } }
      ]"
    >
      <KButton
        appearance="tertiary"
        class="icon-button"
        size="small"
      >

>         <MoreIcon />

Reviewer: question: should the more icon be "built-in" somehow if it's going to be standard?

  <template #footer>
    <div class="horizontal-spacing">
      <KBadge appearance="success">
        Published
      </KBadge>
      <KBadge>2 versions</KBadge>
    </div>
  </template>
</KCard>

```html
<KCard>
  <template #title>
    Card Title
  </template>
  <template #actions>
    <KDropdown
      :items="[
        { label: 'Props', to: { path: '/components/card#props' } },
        { label: 'Slots', to: { path: '/components/card#slots' } }
      ]"
    >
      <KButton
        appearance="tertiary"
        icon
        size="small"
      >
        <MoreIcon />
      </KButton>
    </KDropdown>
  </template>
  Lorem ipsum dolor sit amet ...
  <template #footer>
    <div class="horizontal-spacing">
      <KBadge appearance="success">
        Published
      </KBadge>
      <KBadge>2 versions</KBadge>
    </div>
  </template>
</KCard>
```

## Props

### title

String to be used as a title. Can also be [slotted](#slots).

> ### content

Reviewer: question: does it make sense to keep the `content` prop instead of just requiring the user to slot the card content?

Wondering if we ever use this

String to be used as content. Can also be [slotted](#default).

<KCard
  content="Card content"
  title="Card Title"
/>

## Slots

### title

> Content to be rendered in the title of the card. For example refer to [`footer` slot](#footer).

Reviewer: docs: This is a little confusing to refer to the footer slot for title slot usage

Author: Fixed.

:::warning NOTE

> To preserve a valid HTML structure, avoid slotting in block-level elements such as a `div` into the `title` slot as it will be rendered inside a `h4` element.

Reviewer: question: why do we use an `h4`?

The title should just be a stylized `div` (regardless of whether want them to slot a header tag)

Author: Fixed.

:::

### actions

> Located next to card title, this slot serves for slotting in any action elements. For example refer to [`footer` slot](#footer).

Reviewer: docs: same comment here about referring to the footer slot. Can each slot just have its own example?

Author: Done.

> ### default

Reviewer: suggestion: can we make it a standard to document the default slot first?

Author: Fixed.

> Content of the card. If present, takes precedence over the `content` prop. For example refer to [`footer` slot](#footer). 

Reviewer: Same comment in referring the footer slot example

        icon
        size="small"
      >
        <MoreIcon />
      </KButton>
    </KDropdown>
  </template>
  Lorem ipsum dolor sit amet ...
  <template #footer>
    <KBadge appearance="success">
      Published
    </KBadge>
    <KBadge>2 versions</KBadge>
  </template>
</KCard>
```

## Props

### title

String to be used as a title. Can also be [slotted](#slots).

<KCard title="Card Title" />

```html
<KCard title="Card Title" />
```

## Slots

### default

Content of the card.

<KCard>
  KCard with only default slot content is a container that neatly envelops your content.
</KCard>

```html
<KCard>
  KCard with only default slot content is a container that neatly envelops your content.
</KCard>
```

### title

Content to be rendered in the title of the card.

<KCard>
  <template #title>
    Card Title
  </template>
  When you include a title and content in a KCard, it takes on the appearance of a presentable container for conveying information.
</KCard>

```html
<KCard>
  <template #title>
    Card Title
  </template>
  When you include a title and content in a KCard, it takes on the appearance of a presentable container for conveying information.
</KCard>
```

### actions

> Located next to card title, this slot serves for slotting in any action elements. See the example below.

Reviewer: ```suggestion
Located to the right of the card title, the `actions` slot allows for slotting in any action elements. See the example below.
```
